### PR TITLE
ultralist: rename from todolist, update

### PR DESCRIPTION
**Pull request by uvNikita**
###### Motivation for this change
todolist package was renamed to ultralist.
Here is announcement: ultralist/ultralist#136.

Also updated package to the latest version

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -738,6 +738,9 @@ mapAliases ({
 
   gcc-snapshot = throw "Marked as broken for >2 years, additionally this 'snapshot' pointed to a fairly old one from gcc7.";
 
+  # added 2020-10-13
+  todolist = ultralist;
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:742]: Can you please throw an error?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 100447` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ultralist</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 100447` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ultralist</li>
  </ul>
</details>

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (uvNikita) [reply pkgs/top-level/aliases.nix]: Done

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -738,6 +738,8 @@ mapAliases ({
 
   gcc-snapshot = throw "Marked as broken for >2 years, additionally this 'snapshot' pointed to a fairly old one from gcc7.";
 
+  todolist = throw "todolist is now ultralist.";
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:741]:
```suggestion
  todolist = throw "todolist is now ultralist."; # added 2020-12-27
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 100447` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ultralist</li>
  </ul>
</details>

File: pkgs/applications/misc/ultralist/default.nix [hunk 3]
```diff
@@ -1,21 +1,21 @@
 { stdenv, buildGoPackage, fetchFromGitHub }:
 
 buildGoPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ultralist/default.nix:3]: Please use buildGoModule.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 100447` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ultralist</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
@uvNikita please fix the merge conflict.

File: pkgs/applications/misc/ultralist/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  src = fetchFromGitHub {
+    owner = "ultralist";
+    repo = "ultralist";
+    rev = version;
+    sha256 = "09kgf83jn5k35lyrnyzbsy0l1livzmy292qmlbx5dkdpaq5wxnmp";
+  };
+
+  meta = with lib; {
+    description = "Simple GTD-style todo list for the command line";
+    homepage = "http://ultralist.io";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ultralist/default.nix:18]:
```suggestion
    homepage = "https://ultralist.io";
```

File: pkgs/applications/misc/ultralist/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  version = "1.5.1";
+
+  vendorSha256 = null;
+
+  src = fetchFromGitHub {
+    owner = "ultralist";
+    repo = "ultralist";
+    rev = version;
+    sha256 = "09kgf83jn5k35lyrnyzbsy0l1livzmy292qmlbx5dkdpaq5wxnmp";
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ultralist/default.nix:14]:
```suggestion
  src = fetchFromGitHub {
    owner = "ultralist";
    repo = "ultralist";
    rev = version;
    sha256 = "09kgf83jn5k35lyrnyzbsy0l1livzmy292qmlbx5dkdpaq5wxnmp";
  };

  vendorSha256 = null;
```

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 100447` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>ultralist</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 100447` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>ultralist</li>
</ul>
</details>

**Pull request merged**